crates/re_data_ui/src/rotation3d.rs

```diff
@@ -30,8 +30,11 @@ impl DataUi for datatypes::Rotation3D {
     ) {
         match self {
             Self::Quaternion(q) => {
-                // TODO(andreas): Better formatting for quaternions.
-                ui_layout.data_label(ui, format!("{q:?}"));
+                let [x, y, z, w] = q.xyzw();
+                ui_layout.data_label(
+                    ui,
+                    format!("Quaternion XYZW: [{x:.2} {y:.2} {z:.2} {w:.2}]"),
```

Comment:
That's waaay too wide, means the user definitely has to widen the property view a lot to see the value
<img width="547" alt="image" src="https://github.com/rerun-io/rerun/assets/1220815/284eb364-edcb-43ef-9a6f-a4074363a895">

This problem will go away once we execute on the plan to roll out the rotation/transform unions into components 🤔 . Once we done that, it will say "Quaternion" on the right side.
So we should at the very least put this here 
```TODO(#6388): Once https://github.com/rerun-io/rerun/issues/6388 is solved it should say `Quaternion` on the left side, meaning `Quaternion` shouldn't be in this string```

Reply:
Did you mean to link to the issue about splitting `Rotation3D` into two components (less arrow unions)?

Reply:
> This problem will go away once we execute on the plan to roll out the rotation/transform unions into components 🤔 . Once we done that, it will say "Quaternion" on the right side.

Won't it say `Quaternion` on the _left_ side?